### PR TITLE
Wider vector for FP16 RELU Grad on GPUs

### PR DESCRIPTION
**Pull request by kaixih**
To solve a Build issue brought by https://github.com/tensorflow/tensorflow/pull/39767
@sanjoy

### CODE REVIEW

**Comment by kaixih** [comment]
@sanjoy I fixed one conflict. PTAL.

**Review by sanjoy — COMMENTED** [review]
edit: please ignore; I was somehow looking at just one commit in the PR.

This looks fine to me, but the whole PR was rolled back so you probably need to recreate the previous PR with this fix added on top of it.

**Review by sanjoy — APPROVED** [review]

File: tensorflow/core/kernels/relu_op_gpu.cu.cc [hunk 1]
```diff
@@ -93,6 +94,66 @@ __global__ void ReluGradHalfKernel(const Eigen::half* __restrict__ gradient,
...
+    // Fast path, when half2 primitives are available.
+#if __CUDA_ARCH__ >= 530
+    const half2 kZeroH2 = __float2half2_rn(0.f);
+#endif
+    for (int i = 0; i < VectorSizeElements / 2; i++) {
+#if __CUDA_ARCH__ >= 530
+      // mask = (feature > 0)
+      half2 mask_h2 = __hgt2(feature_h2[i], kZeroH2);
+      // backprop = mask * gradient
+      half2 backprop_h2 = __hmul2(mask_h2, gradient_h2[i]);
```
Comment (akuegel) [inline tensorflow/core/kernels/relu_op_gpu.cu.cc:124]: When using multiplication with the mask, in case the result is 0, the sign is copied from gradient_h2[i]. This makes the results inconsistent with what is computed in line 150. Usually, -0 compares equal to 0, so you would not notice, but for Eigen::half, -0 == 0 doesn't return true.
I am not sure whether those edge cases are important, but wanted to bring this up anyway (I am currently working on adding the MLIR generated ReluGrad kernel, and noticed this mismatch in the test expectations).
